chromium_src/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc

```diff
@@ -52,13 +52,49 @@ String PluginReplacementName(FarblingPRNG* prng) {
   return result.ToString();
 }
 
-void FarblePlugins(DOMPluginArray* owner,
-                   PluginData* data,
+// General helper to add a mime type to a plugin info, with a single extension
+MimeClassInfo* AddMimeTypeToPluginInfo(PluginInfo* plugin_info,
+                                       const String& mime_type,
+                                       const String& description,
+                                       const char* extension) {
+  Vector<String> extensions{String(extension)};
+  auto* mime_info = MakeGarbageCollected<MimeClassInfo>(
+      mime_type, description, *plugin_info, extensions);
+  plugin_info->AddMimeType(mime_info);
+  return mime_info;
+}
+
+// Helper to create a fake plugin with multiple mime types
+DOMPlugin* BraveMakeFakePlugin(blink::LocalDOMWindow* window,
+                               const String& name,
+                               const String& filename,
+                               const String& description,
+                               const std::vector<const char*>& mime_types,
+                               const String& mime_description,
+                               const char* plugin_extension) {
+  auto* plugin_info = MakeGarbageCollected<PluginInfo>(
+      name, filename, description, blink::Color::FromRGBA32(0), false);
+  for (const char* mime_type : mime_types) {
+    AddMimeTypeToPluginInfo(plugin_info, mime_type, mime_description,
+                            plugin_extension);
+  }
+  return MakeGarbageCollected<DOMPlugin>(window, *plugin_info);
+}
+
+DOMPlugin* MakePdfPlugin(const String& plugin_name,
```

Comment:
I'm confused, I don't see where we were doing this before?

Reply:
where is this called from?

Reply:
@emerick but shouldn't it be complaining about unused function here?

Reply:
I feel like maybe it's not complaining because the function was in a `brave` namespace, though I'm not sure. Anyway, it seems unneeded, so removed here https://github.com/brave/brave-core/pull/29767/commits/88e49af90e08213959f44db383c19ee62b19925f.

Reply:
Deleting that is the right thing. Sorry for accidentally leaving it in.

Reply:
No problem at all!